### PR TITLE
HS-52643 support S3 BucketInventory APIs

### PR DESCRIPTION
**Pull request by mozawa**
Added 4 Bucket Inventory APIs to bucket.py and made inventory.py for inventory related classes.
- configure_inventory
- get_inventory_config
- delete_inventory_configuration
- list_inventory_configurations

### CODE REVIEW

File: boto/s3/bucket.py [hunk 1]
```diff
@@ -2451,3 +2452,70 @@ def get_bucket_objectlock_config(self, headers=None):
             response = response.encode('utf-8')
         xml.sax.parseString(response, h)
         return olc
+
+    def configure_inventory(self, id, inventory, headers=None):
```
Comment (tpodowd) [inline boto/s3/bucket.py:2456]: Do other methods in this file not use comments to describe what the method does and the arguments they take?

**Review by tpodowd — CHANGES_REQUESTED** [review]
I think just adding more code docs.

File: boto/s3/bucket.py (hunk 1 above)
Comment (mozawa) [reply boto/s3/bucket.py:2456]: Ok. Added explainations to each method and class to understand what method/class is used for and how those work. Please review again. Thanks.

File: boto/s3/inventory.py [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+
+class InventoryEncryption(object):
+    """
+    Contains the type of server-side encryption used to encrypt
+    the inventory results.
+
+    :ivar kms: Specifies the use of SSE-KMS to encrypt
+        delivered inventory reports.
+
+    :ivar s3: Specifies the use of SSE-S3 to encrypt delivered
```
Comment (tpodowd) [inline boto/s3/inventory.py]: Should this be a boolean? currently you are using None or True. But then lower in to_xml() you are always adding the s3 tag if the value was anything but None. So False would also add the tag. Just wonder if making it a boolean and not having None as a value would help better define it?

**Review by tpodowd — COMMENTED** [review]
Basically looks good. Just one question.

File: boto/s3/inventory.py (hunk 2 above)
Comment (mozawa) [reply boto/s3/inventory.py]: Ok. Updated.

**Review by tpodowd — APPROVED** [review]
Thanks. approving.

**Pull request merged**